ktor-server/ktor-server-core/common/src/io/ktor/server/engine/BaseApplicationResponse.kt

```diff
@@ -179,6 +179,8 @@ public abstract class BaseApplicationResponse(
                 }
             } catch (closed: ClosedWriteChannelException) {
                 throw ChannelWriteException(exception = closed)
+            } finally {
+                flushAndClose()
```

Comment:
TBH, I'm not fully sure if this fixes the original issue (some tests failure), so I will revert this change...
The idea behind this change is that `use` on `ByteWriteChannel` is [custom](https://github.com/ktorio/ktor/blob/3cd270690ca092fd3c01f82e2655821ec587009f/ktor-utils/common/src/io/ktor/util/cio/Readers.kt#L24) and uses `close` which in reality calls `flushAndClose`, but [doesn't wait for it completion](https://github.com/ktorio/ktor/blob/40042dc845e38dcdc2ba2fee0224fbfd2b13d172/ktor-io/common/src/io/ktor/utils/io/ByteWriteChannel.kt#L38).
so `use{}` and `flushAndClose` are mostly the same, except for the fact that `flushAndClose` really suspends

Reply:
reverted in https://github.com/ktorio/ktor/pull/4466/commits/9bbd0d3aa59ff2df9e41df6ece1b780cadb986b2

Now JVM tests are working fine, looks like this was the reason, as this is the only change which affects JVM.
But now `testErrorInBodyClosesConnection` started to fail again - it shows it as flaky, so may be it's really related to this fireAndForget close method in channels

Reply:
@e5l, could you take a look?

Reply:
The difference is in the exception: flushAndClose can throw if the flush is failed. Could you check if this is the case?

Reply:
Do you think may be it will be better to create a separate issue for this to investigate? :)

I see, that the same test fails not only for js/wasm-js. And as I reverted this change, it should not affect anything.

Reply:
Yes, let's move this investigation into a separate task